projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  QueryList,
+  TemplateRef,
+  ViewChild
+} from '@angular/core';
+import { DOWN_ARROW, ENTER, ESCAPE, SPACE, UP_ARROW } from '@angular/cdk/keycodes';
+import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
+import { merge, Subscription } from 'rxjs';
+
+import { KalOptionComponent } from '../kal-option/kal-option.component';
+import { AutoUnsubscribe } from '../../utils/decorators/auto-unsubscribe';
+
+@Component({
+  selector: 'kal-menu',
+  template: `
+    <ng-template>
+      <div
+        class="kal-menu__content"
+        (click)="handleMouse($event)"
+        role="menu">
+        <ng-content></ng-content>
+      </div>
+    </ng-template>`
+})
+export class KalMenuComponent implements AfterContentInit, OnDestroy {
+  @ContentChildren(KalOptionComponent) options: QueryList<KalOptionComponent>;
+
+  @ViewChild(TemplateRef) templateRef: TemplateRef<any>;
+
+
+  /** Event emitted when the menu is closed. */
+  @Output() readonly closed: EventEmitter<void | 'click' | 'keydown' | 'tab' | 'pick'> =
+    new EventEmitter<void | 'click' | 'keydown' | 'tab' | 'pick'>();
+
+  @Output() readonly selectionChange = new  EventEmitter<KalOptionComponent>();
```

Comment:
You emit all KalOptionComponent ? KalOption can have a value

Reply:
to get all information